### ISSUE TITLE
NullPointerException creating executable

### ISSUE THREAD

**Issue by abeckers** [ballerina-platform/ballerina-lang#10956]
**Description:**
NullPointerException creating executable

**Steps to reproduce:**
I have a file mars.bal
```javascript
import ballerina/http;
import congstar/aax_common;
import congstar/aax_contract_list;

endpoint http:Client aax {
    url: "http://localhost:8080/congstar-simulator/services"
};

endpoint http:Listener listener {
    port:9090
};

@http:ServiceConfig {
    basePath: "/"
}
service<http:Service> hello bind listener {
    @http:ResourceConfig {
        methods: ["GET"],
        path: "/contracts"
    }
    contracts(endpoint caller, http:Request request) {
        string customerId = "2012015598";
        var x = aax->post("/contract", aax_contract_list:toRequest(customerId));
        var contract = aax_contract_list:toContracts(x);
        http:Response response = new;
        match contract {
            aax_contract_list:Contract[] res => {
                response.setJsonPayload(check <json>res, contentType = "application/json");
            }
            error err => {
                response.setTextPayload("Error response!\n");
            }
        }
        _ = caller -> respond(response);
    }
}
```
a file aax_contract_list/aax-contract-list.bal
```javascript
import ballerina/http;
import ballerina/io;
import congstar/aax_common;

public type Contract record {
    string contractNumber;
};

public function toRequest(string customerId) returns xml {
    xmlns "http://abc";
    xml b = xml `<ContractList><customerID>{{customerId}}</customerID></ContractList>`;
    return b;
}

public function toContracts(http:Response|error res) returns Contract[]|error {
    match res {
        http:Response r => {
            xml body = check r.getXmlPayload();
            Contract[] c = [];
            return c;
        }
        error err => {
            return err;
        }
    }
}
```
and a file aax_common/util.bal
```javascript
public function aaxheader() returns xml {
    xmlns "http://abc";
    return xml `<header><source>9</source><version>1.11</version><client>1</client></header>`;
}
```
when I invoke the builder `ballerina build` I get
>> ballerina: Oh no, something really went wrong. Bad. Sad.

>> There should be a file named "ballerina-internal.log" in the current directory.
>> If you are able to share with us the code that broke Ballerina then
>> we would REALLY appreciate if you would report this to us:
>> go to https://github.com/ballerina-platform/ballerina-lang/issues and
>> create a bug report with both this log file and the sample code.

>> We thank you for helping make us better dancers.

And in the log:
```
[2018-10-23 08:00:17,814] ERROR {org.ballerinalang.launcher.Main} - null 
java.lang.NullPointerException
        at org.wso2.ballerinalang.programfile.ProgramFileWriter.writePackageInfoEntries(ProgramFileWriter.java:72)
        at org.wso2.ballerinalang.programfile.ProgramFileWriter.writeProgram(ProgramFileWriter.java:46)
        at org.wso2.ballerinalang.compiler.BinaryFileWriter.writeExecutableBinary(BinaryFileWriter.java:112)
        at org.wso2.ballerinalang.compiler.BinaryFileWriter.writeExecutableBinary(BinaryFileWriter.java:102)
        at org.wso2.ballerinalang.compiler.BinaryFileWriter.write(BinaryFileWriter.java:86)
        at java.util.ArrayList.forEach(ArrayList.java:1257)
        at org.wso2.ballerinalang.compiler.Compiler.write(Compiler.java:111)
        at org.ballerinalang.packerina.BuilderUtils.compileWithTestsAndWrite(BuilderUtils.java:106)
        at org.ballerinalang.packerina.cmd.BuildCommand.execute(BuildCommand.java:95)
        at java.util.Optional.ifPresent(Optional.java:159)
        at org.ballerinalang.launcher.Main.main(Main.java:65)
```

Removing the line
```import congstar/aax_common;```
from the mars.bal file makes the NPE disappear.

**Affected Versions:**
last weeks and today nightly build

**OS, DB, other environment details and versions:**
MacOS

**Comment by abeckers** [comment]
seems also to occur if i have
modules A, B, C, D
A imports B and C
B and C import D

**Comment by abeckers** [comment]
Oh, there seem to be two problems. This second problem leads to a different error message while compiling the source:
```
[2018-10-23 09:09:59,978] ERROR {org.ballerinalang.launcher.Main} - null 
java.lang.StackOverflowError
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TaintAnalyzer.visit(TaintAnalyzer.java:1041)
        at org.wso2.ballerinalang.compiler.tree.expressions.BLangXMLElementLiteral.accept(BLangXMLElementLiteral.java:111)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TaintAnalyzer.visit(TaintAnalyzer.java:1091)
        at org.wso2.ballerinalang.compiler.tree.expressions.BLangXMLElementLiteral.accept(BLangXMLElementLiteral.java:111)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TaintAnalyzer.analyzeNode(TaintAnalyzer.java:1361)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TaintAnalyzer.visit(TaintAnalyzer.java:388)
        at org.wso2.ballerinalang.compiler.tree.BLangVariable.accept(BLangVariable.java:88)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TaintAnalyzer.visit(TaintAnalyzer.java:440)
        at org.wso2.ballerinalang.compiler.tree.statements.BLangVariableDef.accept(BLangVariableDef.java:42)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TaintAnalyzer.analyzeNode(TaintAnalyzer.java:1361)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TaintAnalyzer.visit(TaintAnalyzer.java:433)
        at org.wso2.ballerinalang.compiler.tree.statements.BLangBlockStmt.accept(BLangBlockStmt.java:54)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TaintAnalyzer.analyzeNode(TaintAnalyzer.java:1361)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TaintAnalyzer.analyzeReturnTaintedStatus(TaintAnalyzer.java:1639)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TaintAnalyzer.analyzeReturnTaintedStatus(TaintAnalyzer.java:1588)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TaintAnalyzer.analyzeAllParamsUntaintedReturnTaintedStatus(TaintAnalyzer.java:1564)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TaintAnalyzer.visitInvokable(TaintAnalyzer.java:1519)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TaintAnalyzer.visit(TaintAnalyzer.java:334)
        at org.wso2.ballerinalang.compiler.tree.BLangFunction.accept(BLangFunction.java:66)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TaintAnalyzer.resolveBlockedInvokable(TaintAnalyzer.java:2051)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TaintAnalyzer.resolveBlockedInvokable(TaintAnalyzer.java:2065)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TaintAnalyzer.resolveBlockedInvokable(TaintAnalyzer.java:2065)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TaintAnalyzer.resolveBlockedInvokable(TaintAnalyzer.java:2065)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TaintAnalyzer.resolveBlockedInvokable(TaintAnalyzer.java:2065)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TaintAnalyzer.resolveBlockedInvokable(TaintAnalyzer.java:2065)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TaintAnalyzer.resolveBlockedInvokable(TaintAnalyzer.java:2065)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TaintAnalyzer.resolveBlockedInvokable(TaintAnalyzer.java:2065)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TaintAnalyzer.resolveBlockedInvokable(TaintAnalyzer.java:2065)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TaintAnalyzer.resolveBlockedInvokable(TaintAnalyzer.java:2065)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TaintAnalyzer.resolveBlockedInvokable(TaintAnalyzer.java:2065)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TaintAnalyzer.resolveBlockedInvokable(TaintAnalyzer.java:2065)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TaintAnalyzer.resolveBlockedInvokable(TaintAnalyzer.java:2065)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TaintAnalyzer.resolveBlockedInvokable(TaintAnalyzer.java:2065)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TaintAnalyzer.resolveBlockedInvokable(TaintAnalyzer.java:2065)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TaintAnalyzer.resolveBlockedInvokable(TaintAnalyzer.java:2065)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TaintAnalyzer.resolveBlockedInvokable(TaintAnalyzer.java:2065)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TaintAnalyzer.resolveBlockedInvokable(TaintAnalyzer.java:2065)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TaintAnalyzer.resolveBlockedInvokable(TaintAnalyzer.java:2065)
...
```

**Comment by ayomawdb** [comment]
@abeckers Yes, this should be related to #10905. To be sure, I'll check your code with the changes suggested in the PR and update this issue. Thank you!

**Comment by ayomawdb** [comment]
@abeckers Sorry about the delay in response. This issue was fixed with the PR #10905. I have tested your code with ballerina-0.983.0 RC2 pack and code compiles as expected.

**Comment by ayomawdb** [comment]
Resolved with PR #10905
